### PR TITLE
Jbrun/tdp 2460 migrate pe data in ee

### PR DESCRIPTION
**Pull request by hmdebenque**
https://jira.talendforge.org/browse/TDP-2460

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

### CODE REVIEW

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 1]
```diff
@@ -170,13 +173,19 @@ public String create(final Preparation preparation, String folderId) {
...
+     * @param name
      * @param sort how the preparation should be sorted (default is 'last modification date').
      * @param order how to apply the sort.
      * @return the preparations id list.
      */
-    public Stream<String> list(Sort sort, Order order) {
+    public Stream<String> list(String name, Sort sort, Order order) {
         LOGGER.debug("Get list of preparations (summary).");
-        return preparationRepository.list(Preparation.class) //
+        Stream<Preparation> preparationStream = preparationRepository.list(Preparation.class);
```
Comment (fhuaulme) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: Better use org.talend.dataprep.preparation.store.PreparationRepository#list(java.lang.Class<T>, java.lang.String) with a TQL filter.

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 2]
```diff
@@ -170,13 +173,19 @@ public String create(final Preparation preparation, String folderId) {
     /**
      * List all the preparations id.
      *
+     *
+     * @param name
```
Comment (fhuaulme) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: Missing parameter Javadoc

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 3]
```diff
@@ -189,9 +198,13 @@ public String create(final Preparation preparation, String folderId) {
      * @param order how to order the sort.
      * @return the preparation details.
      */
-    public Stream<UserPreparation> listAll(Sort sort, Order order) {
+    public Stream<UserPreparation> listAll(String name, Sort sort, Order order) {
         LOGGER.debug("Get list of preparations (with details).");
-        return preparationRepository.list(Preparation.class) //
+        Stream<Preparation> preparationStream = preparationRepository.list(Preparation.class);
```
Comment (fhuaulme) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: Better use org.talend.dataprep.preparation.store.PreparationRepository#list(java.lang.Class, java.lang.String) with a TQL filter.

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 4]
```diff
@@ -201,9 +214,13 @@ public String create(final Preparation preparation, String folderId) {
      *
      * @return the preparation summaries, sorted by descending last modification date.
      */
-    public Stream<PreparationSummary> listSummary() {
+    public Stream<PreparationSummary> listSummary(String name) {
         LOGGER.debug("Get list of preparations (summary).");
-        return preparationRepository.list(Preparation.class) //
+        Stream<Preparation> preparationStream = preparationRepository.list(Preparation.class);
+        if (name != null) {
```
Comment (fhuaulme) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: Better use org.talend.dataprep.preparation.store.PreparationRepository#list(java.lang.Class, java.lang.String) with a TQL filter.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/552/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/552/)

**Pull request merged**